draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2931,253 @@ processing the server's Certificate, CertificateVerify, and
+{:br }
+
+Note: Prior to TLS 1.3, "certificate_list" ordering required each certificate
+to certify the one immediately preceding it,
+however some implementations allowed some flexibility. Servers sometimes send
+both a current and deprecated intermediate for transitional purposes, and others
+are simply configured incorrectly, but these cases can nonetheless be validated
+properly. For maximum compatibility, all implementations SHOULD be prepared to
+handle potentially extraneous certificates and arbitrary orderings from any TLS
+version, with the exception of the end-entity certificate which MUST be first.
+
+The server's certificate list MUST always be non-empty. A client MAY
+MAY send no certificates if it does not have an appropriate
+certificate to send in response to the server's authentication
+request.
+
+##### Server Certificate Selection
+
+The following rules apply to the certificates sent by the server:
+
+-  The certificate type MUST be X.509v3 {{RFC5280}}, unless explicitly negotiated
+  otherwise (e.g., {{RFC5081}}).
+
+-  The server's end-entity certificate's public key (and associated
+  restrictions) MUST be compatible with the selected key exchange
+  algorithm.
+
+~~~~
+    Key Exchange Alg.  Certificate Key Type
+
+    DHE_RSA            RSA public key; the certificate MUST allow the
+    ECDHE_RSA          key to be used for signing (i.e., the
+                       digitalSignature bit MUST be set if the key
+                       usage extension is present) with the signature
+                       scheme and hash algorithm that will be employed
+                       in the server's KeyShare extension.
+                       Note: ECDHE_RSA is defined in [RFC4492].
+
+    ECDHE_ECDSA        ECDSA-capable public key; the certificate MUST
+                       allow the key to be used for signing with the
+                       hash algorithm that will be employed in the
+                       server's KeyShare extension.  The public key
+                       MUST use a curve and point format supported by
+                       the client, as described in [RFC4492].
+~~~~
+
+- The "server_name" and "trusted_ca_keys" extensions {{RFC6066}} are used to
+  guide certificate selection. As servers MAY require the presence of the server_name
+  extension, clients SHOULD send this extension.
+
+All certificates provided by the server MUST be signed by a
+hash/signature algorithm pair that appears in the "signature_algorithms"
+extension provided by the client, if they are able to provide such
+a chain (see {{signature-algorithms}}).
+If the server cannot produce a certificate chain that is signed only via the
+indicated supported pairs, then it SHOULD continue the handshake by sending
+the client a certificate chain of its choice that may include algorithms
+that are not known to be supported by the client. This fallback chain MAY
+use the deprecated SHA-1 hash algorithm.
```

Comment:
... if the client indicated that it supported SHA-1 in combination with the signature algorithm in "signature_algorithms".